vcd/resource_vcd_vapp_vm.go

```diff
@@ -450,11 +450,14 @@ func resourceVcdVAppVmUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 
 		if d.Get("power_on").(bool) {
-			task, err := vm.PowerOn()
-			if err != nil {
-				return fmt.Errorf("error Powering Up: %#v", err)
-			}
-			err = task.WaitTaskCompletion()
+			err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
```

Comment:
Please add a comment on why this retry is needed.

Reply:
I added generic comment for all retryCall. 
```
// all functionality using retryCall as multi requests to same vAPP isn't handled and errors occur
```